packages/ui/src/composables/useThrottle.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @description returns throttled function or value
+ * the last one always returns last-call value in the end if no more new calls were provided
+ * @example
+ *    import { useThrottleFunction, useThrottleValue } from '../../composables'
+ *    ...
+ *    const localThrottledFunction = useThrottleFunction(functionToThrottle, props)
+ *    const localThrottledValue = useThrottleValue(reactiveValueToThrottle, props)
+ */
+
+import {
+  ref, toRef, unref,
+  watch,
+  Ref, ExtractPropTypes, ComponentInternalInstance,
+} from 'vue'
+
+type ThrottledFunctionArgs = any[]
+type ThrottledFunction<Output> = (...args: ThrottledFunctionArgs) => Output
+
+export const useThrottleProps = {
+  delay: {
+    type: Number,
+    default: 0,
+    validator: (value: number) => value >= 0,
+  },
+}
+
+type UseThrottleProps = ExtractPropTypes<typeof useThrottleProps>
+
+/**
+ * @param fn passed function to throttle
+ * @param props { delay } call delay in ms
+ */
+export function useThrottleFunction<Output> (fn: ThrottledFunction<Output>, props: UseThrottleProps) {
```

Comment:
Because it is a composable, it would be correctly to pass `Ref<number>` as dealy instead of props.

Reply:
And you're suggesting not to reuse useThrottleProps?